### PR TITLE
Fix Pylint warnings for the new checkers

### PR DESCRIPTION
**Pull request by ostoliarova-msft**
# Description

- Renamed enums to be uppercase
- Removed disabling of lint rules for addressed items
- Added a keyword-only api_version argument:  ":keyword api_version:" to client constructors

For more information, please check out the [issue ](https://github.com/Azure/azure-sdk-for-python/issues/24146)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by elavarasidc** [comment]
The changes, look fine to me.
Adding in @shirleyqin-msft to pull in the changes locally, validate and sign off for Rooms.

cc: @radubulboaca

**Review by shirleyqin-msft — APPROVED** [review]

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py
Comment (ostoliarova-msft) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py:64]: Chat team, is this change fine here? Method was regenerated with a new signature

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py
Comment (AikoBB) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py:47]:
```suggestion
    :keyword api_version: Azure Communication Chat API Version. Default value is "2021-09-07". Note that overriding this
```

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py
Comment (AikoBB) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py:63]:
```suggestion
    :keyword api_version: Azure Communication Chat API Version. Default value is "2021-09-07". Note that overriding this
```

File: sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_client_async.py
Comment (AikoBB) [inline sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_client_async.py:52]:
```suggestion
    :keyword api_version: Azure Communication Chat API. Default value is "2021-09-07". Note that overriding this
```

File: sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_thread_client_async.py
Comment (AikoBB) [inline sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_thread_client_async.py:63]:
```suggestion
    :keyword api_version: Azure Communication Chat API. Default value is "2021-09-07". Note that overriding this
```

File: sdk/communication/azure-communication-chat/swagger/SWAGGER.md
Comment (AikoBB) [inline sdk/communication/azure-communication-chat/swagger/SWAGGER.md:20]: why did we change the commit hash?

File: sdk/communication/azure-communication-email/azure/communication/email/_email_client.py
Comment (AikoBB) [inline sdk/communication/azure-communication-email/azure/communication/email/_email_client.py:29]:
```suggestion
    :keyword api_version: Azure Communication Email API . Default value is "2021-10-01-preview". Note that overriding
```

File: sdk/communication/azure-communication-email/azure/communication/email/aio/_email_client_async.py
Comment (AikoBB) [inline sdk/communication/azure-communication-email/azure/communication/email/aio/_email_client_async.py:28]:
```suggestion
    :keyword api_version: Azure Communication Email API . Default value is "2021-10-01-preview". Note that overriding
```

File: sdk/communication/azure-communication-identity/azure/communication/identity/_communication_identity_client.py
Comment (AikoBB) [inline sdk/communication/azure-communication-identity/azure/communication/identity/_communication_identity_client.py:31]:
```suggestion
    :keyword api_version: Azure Communication Identity API version. Default value is "2022-06-01". Note that overriding this
```

File: sdk/communication/azure-communication-identity/azure/communication/identity/aio/_communication_identity_client_async.py
Comment (AikoBB) [inline sdk/communication/azure-communication-identity/azure/communication/identity/aio/_communication_identity_client_async.py:30]:
```suggestion
    :keyword api_version: Azure Communication Identity API version. Default value is "2022-06-01". Note that overriding this
```

File: sdk/communication/azure-communication-networktraversal/azure/communication/networktraversal/_communication_relay_client.py
Comment (AikoBB) [inline sdk/communication/azure-communication-networktraversal/azure/communication/networktraversal/_communication_relay_client.py:31]:
```suggestion
    :keyword api_version: Azure Communication Network Traversal API version. Default value is "2022-03-01-preview". Note that overriding
```

File: sdk/communication/azure-communication-networktraversal/azure/communication/networktraversal/aio/_communication_relay_client_async.py
Comment (AikoBB) [inline sdk/communication/azure-communication-networktraversal/azure/communication/networktraversal/aio/_communication_relay_client_async.py:30]:
```suggestion
    :keyword api_version: Azure Communication Network Traversal API version. Default value is "2022-03-01-preview". Note that overriding
```

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py
Comment (AikoBB) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py:32]:
```suggestion
    :keyword api_version: Azure Communication Phone Number API version. The default value is "2022-01-11-preview2". Note that
```

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/aio/_phone_numbers_client_async.py
Comment (AikoBB) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/aio/_phone_numbers_client_async.py:33]:
```suggestion
    :keyword api_version: Azure Communication Phone Number API version. The default value is "2022-01-11-preview2". Note that
```

File: sdk/communication/azure-communication-rooms/azure/communication/rooms/_rooms_client.py
Comment (AikoBB) [inline sdk/communication/azure-communication-rooms/azure/communication/rooms/_rooms_client.py:41]:
```suggestion
    :keyword api_version: Azure Communication Rooms API version. Default value is "2022-02-01". Note that overriding this
```

File: sdk/communication/azure-communication-rooms/azure/communication/rooms/aio/_rooms_client_async.py
Comment (AikoBB) [inline sdk/communication/azure-communication-rooms/azure/communication/rooms/aio/_rooms_client_async.py:37]:
```suggestion
    :keyword api_version: Azure Communication Rooms API version. Default value is "2022-02-01". Note that overriding this
```

File: sdk/communication/azure-communication-sms/azure/communication/sms/_sms_client.py
Comment (AikoBB) [inline sdk/communication/azure-communication-sms/azure/communication/sms/_sms_client.py:30]:
```suggestion
    :keyword api_version: Azure Communication SMS API version. The default value is "2021-03-07". Note that
```

File: sdk/communication/azure-communication-sms/azure/communication/sms/aio/_sms_client_async.py
Comment (AikoBB) [inline sdk/communication/azure-communication-sms/azure/communication/sms/aio/_sms_client_async.py:30]:
```suggestion
    :keyword api_version: Azure Communication SMS API version. The default value is "2021-03-07". Note that
```

**Comment by AikoBB** [comment]
@ostoliarova-msft pls apply the pylint changes for the [azure-communication-jobrouter](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/communication/azure-communication-jobrouter) as well

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py
Comment (AikoBB) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py:64]: @LuChen-Microsoft need your confirmation here

**Comment by AikoBB** [comment]
> **API change check**
> 
> APIView has identified API level changes in this PR and created following API reviews.
> 
> [azure-communication-identity](https://apiview.dev/Assemblies/Review/37a7defbbc284d3cb58f78f5e3b4fdbb) [azure-communication-phonenumbers](https://apiview.dev/Assemblies/Review/6bf7ca79a0f148248ac1107fbcf4ca46) [azure-communication-networktraversal](https://apiview.dev/Assemblies/Review/9c2445b7579a4b9e9390b19944bec371) [azure-communication-rooms](https://apiview.dev/Assemblies/Review/a3a22b8c0f994bfa87023ca6668e4227) [azure-communication-chat](https://apiview.dev/Assemblies/Review/6af9f4d070624f8eb817c799082f3b3c) [azure-communication-sms](https://apiview.dev/Assemblies/Review/e058cc51fec24e428cfd4e1fba717017)

@annatisch this PR includes implementation of [Fix Pylint warnings for the new checkers ](https://github.com/Azchaure/azure-sdk-for-python/issues/24146) issue and we are introducing the following API changes, I was wondering if these changes should be released as a minor version release

File: sdk/communication/azure-communication-chat/swagger/SWAGGER.md
Comment (ostoliarova-msft) [reply sdk/communication/azure-communication-chat/swagger/SWAGGER.md:20]: @AikoBB I made it in correspondence with other modalities where swagger is generated based on the specific commit and not on the branch.  Do I need to rollback this change and regenerate from swagger in `main`?

File: sdk/communication/azure-communication-chat/azure/communication/chat/_api_versions.py
Comment (ostoliarova-msft) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_api_versions.py:11]: [_azure_communication_chat_service](https://github.com/ostoliarova-msft/azure-sdk-for-python/blob/ostoliarova/sdk-communication/pylint_warnings/sdk/communication/azure-communication-chat/azure/communication/chat/_generated/_azure_communication_chat_service.py#L33) lists "2021-09-07" as default version. However, there is no such version in the link provided by @AikoBB (there are versions "2021-**03**-07" and "2021-09-**27**"). Could you please advise, what versions should be included in _api_versions.py?

Comment (AikoBB) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_api_versions.py:11]: @LuChen-Microsoft

File: sdk/communication/azure-communication-chat/swagger/SWAGGER.md
Comment (AikoBB) [reply sdk/communication/azure-communication-chat/swagger/SWAGGER.md:20]: It's up to the chat team, if they want to go with the main branch then it's better to revert as it was. @LuChen-Microsoft need your approval here as well :)

**Review by danielortega-msft — APPROVED** [review]

**Comment by jiriburant** [comment]
Failing tests for sip routing shouldnt be a blocker, it's a problem with test data that we're going to fix.

**Review by AikoBB — APPROVED** [review]
Thank you so much for fixing it for the whole ACS modalities! Once it's merged into the main branch pls get approval from Anna Tisch regarding the apivew changes

**Review by yogeshmo — APPROVED** [review]
The email changes look good to me - thanks!

**Review by sarkar-rajarshi — APPROVED** [review]
Looks for job router

**Comment by AikoBB** [comment]
@ostoliarova-msft one last thing, pls mention that we are providing the ability specify the API version by `api_version` keyword param in the changelog files under the `Features Added` section for each modality

**Review by petrsvihlik — APPROVED** [review]

**Review by radubulboaca — APPROVED** [review]

**Pull request merged**
